dvc/remote/ssh/connection.py

```diff
@@ -143,7 +144,16 @@ def walk_files(self, directory):
 
     def remove(self, path):
         self._sftp_connect()
-        self._sftp.remove(path)
+        try:
+            self._sftp.remove(path)
+        except FileNotFoundError as exc:
```

Comment:
@mroutis it's not Python 2 compatible, right? is true for sftp client as well that it throws IoError on python 2 instead?

Reply:
Yes, we should use IOError and check for exc.errno. That would work for both py2 and py3.